Assets/LeapMotion/Core/Scripts/Hands/HandDataShaderSupport.cs

```diff
@@ -0,0 +1,323 @@
+
+    public const string LEFT_FINGER_SEGMENTS_UNIFORM_NAME  = "_Leap_LH_FingerSegments";
+    public const string RIGHT_FINGER_SEGMENTS_UNIFORM_NAME = "_Leap_RH_FingerSegments";
+
+    [SerializeField, Disable]
+    private int _leftFingertipsUniformParamId = 0;
+    [SerializeField, Disable]
+    private int _rightFingertipsUniformParamId = 0;
+
+    [SerializeField, Disable]
+    private int _leftPalmTrisUniformParamId = 0;
+    [SerializeField, Disable]
+    private int _rightPalmTrisUniformParamId = 0;
+
+    [SerializeField, Disable]
+    private int _leftFingerSegmentsUniformParamId = 0;
+    [SerializeField, Disable]
+    private int _rightFingerSegmentsUniformParamId = 0;
+
+    private Vector4[] _leftFingertips = new Vector4[5];
+    private Vector4[] _rightFingertips = new Vector4[5];
+
+    private Vector4[] _leftPalmTris = new Vector4[15];
+    private Vector4[] _rightPalmTris = new Vector4[15];
+
+    private Vector4[] _leftFingerSegments = new Vector4[28];
+    private Vector4[] _rightFingerSegments = new Vector4[28];
+
+    private void OnValidate() {
+      refreshParamIDs();
+
+      uploadEditTimeFingerAtTransform();
+    }
+
+    private void Reset() {
+      if (provider == null) provider = Hands.Provider;
+    }
+
+    private void Start() {
+      if (provider == null) provider = Hands.Provider;
+
+      refreshParamIDs();
+    }
+
+    private void refreshParamIDs() {
+      _leftFingertipsUniformParamId = Shader.PropertyToID(LEFT_FINGERTIPS_UNIFORM_NAME);
+      _rightFingertipsUniformParamId = Shader.PropertyToID(RIGHT_FINGERTIPS_UNIFORM_NAME);
+
+
+      _leftPalmTrisUniformParamId = Shader.PropertyToID(LEFT_PALM_TRIANGLES_UNIFORM_NAME);
+      _rightPalmTrisUniformParamId = Shader.PropertyToID(RIGHT_PALM_TRIANGLES_UNIFORM_NAME);
+
+
+      _leftFingerSegmentsUniformParamId = Shader.PropertyToID(LEFT_FINGER_SEGMENTS_UNIFORM_NAME);
+      _rightFingerSegmentsUniformParamId = Shader.PropertyToID(RIGHT_FINGER_SEGMENTS_UNIFORM_NAME);
+    }
+
+    private void uploadEditTimeFingerAtTransform() {
+      _leftFingertips[0] = this.transform.position.WithW(1);
+      _rightFingertips[0] = this.transform.position.WithW(1);
+      Shader.SetGlobalVectorArray(_leftFingertipsUniformParamId, _leftFingertips);
+      Shader.SetGlobalVectorArray(_rightFingertipsUniformParamId, _rightFingertips);
+    }
+
+    private void Update() {
+
+      var leftHand = provider.CurrentFrame.Hands.Query()
+                     .FirstOrDefault(h => h.IsLeft);
+      if (leftHand != null) {
+
+        // Fingertips
+        for (int i = 0; i < NUM_FINGERS; i++) {
+          _leftFingertips[i] = leftHand.Fingers[i].TipPosition.ToVector3().WithW(1);
+        }
+
+        // Palm Tris
+        {
+          var hand = leftHand;
+          var palmTris = _leftPalmTris;
+          int i = 0;
+
+          // Left Hand Palm Tris
+          {
+            Finger thumb = hand.GetThumb(), index = hand.GetIndex(),
```

Comment:
I think I might have made a comment here before, maybe a diff erased it?  This looks like a lot of duplicate code for left and right, would love it if they could be combined.